orchestra/analytics/latency.py

```diff
@@ -0,0 +1,112 @@
+import itertools
+from datetime import timedelta
+from dateutil.parser import parse
+from operator import attrgetter
+from orchestra.models import Task
+from orchestra.project_api.api import get_project_information
+from orchestra.workflow import Step
+from pandas import DataFrame
+
+
+def work_time_df(projects):
+    """
+    Return all projects' task assignment iteration timing information as a
+    dataframe.
+
+    The table schema is:
+        project: the project's ID
+        task: the task's ID
+        assignment: a 0-indexed assignment counter (0 is entry-level, 1 is
+            first review, etc.)
+        iteration: the work iteration for that assignment.
+            - assignment 0 iteration 0 represents the period from task creation
+              to first assignment.
+            - assignment N > 0 iteration 0 represents the period from previous
+              submission to reviewer assignment.
+        start_datetime: the time the iteration started.
+        calendar_time: the wallclock time it took for the worker to complete
+            that iteration.
+        work_time: the amount of work time the worker reported dedicating to
+            that iteration (0 for iteration 0).
+
+    Args:
+        projects (django.db.models.query.QuerySet):
+            An iterable of projects to include in the report.
+
+    Returns:
+        df (pandas.DataFrame):
+            A DataFrame with timing information.
+    """
+    # TODO(marcua): `get_project_information` per project means we do random
+    # access all over the database.  Ideally, we'd have a sequential scan
+    # version of this.
+    time_data = list(itertools.chain.from_iterable(
+        project_time_row_generator(
+            get_project_information(project.id))
+        for project in projects))
+    df = DataFrame(time_data)
+    df = df[
+        ['project', 'task', 'assignment', 'iteration', 'start_datetime',
+         'calendar_time', 'work_time']]
```

Comment:
Ah, ok. Should that happen in the analysis script? (feel free to say no and leave it be).